api/api.go

```diff
@@ -22,6 +24,36 @@ type Request struct {
 	Mixins []string
 	// The map of parameters to use.
 	Params map[string][]string
+	// The type of request (i.e. GET/POST etc.)
+	ReqType HTTPMethod
+	// The body of the request
+	Body bytes.Buffer
+}
+
+// HTTPMethod guards against incorrect methods being specified through strings
+type HTTPMethod int
+
+const (
```

Comment:
This goes against what @MattWindsor91 said, but maybe golang's builtins would be better? https://golang.org/pkg/net/http/#pkg-constants still strings, but less boilerplate that has do be done here

Reply:
I did try methods sticking with strings, but it doesn't add any security for future methods.
eg if I write another post method, I can still just write `"POST"` and not even know it's changed, whereas this method _forces_ the programmer to use a valid request type.